src/base/QXmppElement.cpp

```diff
@@ -123,10 +123,13 @@ QXmppElement::~QXmppElement()
 
 QXmppElement &QXmppElement::operator=(const QXmppElement &other)
 {
-    other.d->counter.ref();
-    if (!d->counter.deref())
-        delete d;
-    d = other.d;
+    if (this != &other) // self-assignment check
```

Comment:
> Also, note that, say, typical std::string implementations don't do any self-assignment checks in their operator= (at least, I'm not aware of any implementations doing so).

Why have you mentioned `std::string`? I do not know about it, but other STL classes do have self-assignment checks in `operator=`. At least in libstdc++.